Tools/autotest/arducopter.py

```diff
@@ -1129,6 +1132,89 @@ def fly_mission(self, height_accuracy=-1.0, target_altitude=None):
+
+        old_pos = self.mav.recv_match(type='GLOBAL_POSITION_INT', blocking=True)
+        print("old_pos=%s" % str(old_pos))
+
+        old_gps_type = self.get_parameter("GPS_TYPE")
+        old_ek2_gps_type = self.get_parameter("EK2_GPS_TYPE")
+        old_serial5_protocol = self.get_parameter("SERIAL5_PROTOCOL")
+
+        ex = None
+        try:
+            self.set_parameter("GPS_TYPE", 0)
+            self.set_parameter("EK2_GPS_TYPE", 3)
+            self.set_parameter("SERIAL5_PROTOCOL", 1)
+            self.reboot_sitl()
+            # without a GPS or some sort of external prompting, AP
+            # doesn't send system_time messages.  So prompt it:
+            self.mav.mav.system_time_send(time.time() * 1000000, 0)
+            self.mav.mav.set_gps_global_origin_send(1,
+                                                    old_pos.lat,
+                                                    old_pos.lon,
+                                                    old_pos.alt)
+            self.progress("Waiting for non-zero-lat")
+            while True:
+                gpi = self.mav.recv_match(type='GLOBAL_POSITION_INT',
+                                          blocking=True)
+#                self.progress("gpi=%s" % str(gpi))
+                if gpi.lat != 0:
+                    break
+
+            self.takeoff(arm=True)
+            self.set_rc(1, 1600)
+            while True:
+                vicon_pos = self.mav.recv_match(type='VICON_POSITION_ESTIMATE',
+                                                blocking=True)
+#                print("vpe=%s" % str(vicon_pos))
+                gpi = self.mav.recv_match(type='GLOBAL_POSITION_INT',
+                                          blocking=True)
+#                self.progress("gpi=%s" % str(gpi))
+                if vicon_pos.x > 40:
+                    break
+            # recenter controls:
+            self.set_rc(1, 1500)
+            self.progress("# Enter RTL")
+            self.mavproxy.send('switch 3\n')
+            self.set_rc(3, 1500)
+            tstart = self.get_sim_time()
+            while True:
+                if self.get_sim_time() - tstart > 200:
+                    raise NotAchievedException()
+                gpi = self.mav.recv_match(type='GLOBAL_POSITION_INT',
+                                          blocking=True)
+#                print("gpi=%s" % str(gpi))
+                ss = self.mav.recv_match(type='SIMSTATE',
+                                         blocking=True)
+#                print("ss=%s" % str(ss))
+                # wait for RTL disarm:
+                if not self.armed():
+                    break
+
+        except Exception as e:
+            self.progress("Exception caught")
+            ex = e
+
+        self.set_parameter("GPS_TYPE", old_gps_type)
+        self.set_parameter("EK2_GPS_TYPE", old_ek2_gps_type)
+        self.set_parameter("SERIAL5_PROTOCOL", old_serial5_protocol)
+        self.set_rc(3, 1000)
```

Comment:
I don't understand the RC3 changes, after RTL you set it to 1500 and here (where we are either landed and disarmed or armed somewhere), you it set to 1000. Can you explain the rationale?

Reply:
This mimics what a pilot should be doing on the throttle stick.  When in an auto-throttle mode having your stick at half is a sensible thing to do in case you need to switch to a manual-throttle mode.

Setting it to 1000 when disarmed means the next poor sucker can arm :-)